decidim-core/lib/tasks/decidim_locales_tasks.rake

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+namespace :decidim do
+  namespace :locales do
+    desc "Ensures locales in organizations are in sync with Decidim initializer"
+    task sync_all: :environment do
+      Decidim::Organization.all.each do |organization|
+        organization.available_locales = Decidim.available_locales.filter do |lang|
+          organization.available_locales.include?(lang.to_s)
+        end
+        organization.default_locale = organization.available_locales.first unless organization.available_locales.include? organization.default_locale
+        organization.save!
+      end
+    end
+
+    desc "Rebuild the search index"
+    task rebuild_search: :environment do
+      Decidim::SearchableResource.destroy_all
+      Decidim::Searchable.searchable_resources.keys.each do |resource|
+        resource.constantize.all.each(&:try_update_index_for_search_resource)
```

Comment:
```suggestion
        resource.constantize.find_each(&:try_update_index_for_search_resource)
```

This is more performant, memory-wise, as it loads the resources in chunks instead of all of them at the same time.

Reply:
Also, the `Decidim::Searchable.searchable_resources` has the classes as values. I'm not sure if you're constantizing for some specific reason, but the whole block could be changed to this:

```ruby
      Decidim::Searchable.searchable_resources.values.each do |searchable_klass|
        searchable_klass.find_each(&:try_update_index_for_search_resource)
      end
```

Reply:
oh, I don't even remember. I guess I saw the key and my brain executed a `return`!
Thanks for your insights! I'll make the changes!

Reply:
If you prefer, you can just ignore these lastc hangees and keep only the `find_each` suggestion 😄

Reply:
No, I like it. I'll continue in the afternoon, I have a busy morning.
Thinking out loud, we could add a task to add or remove a language from an organization and perform all the associated task for that organization.